calico_containers/tests/st/utils/utils.py

```diff
@@ -20,26 +20,37 @@
 from exceptions import CommandExecError
 import re
 import json
+from pycalico.util import get_host_ips
 
 LOCAL_IP_ENV = "MY_IP"
+LOCAL_IPv6_ENV = "MY_IPv6"
 logger = logging.getLogger(__name__)
 
 
-def get_ip():
+def get_ip(v6=False):
     """
     Return a string of the IP of the hosts interface.
     Try to get the local IP from the environment variables.  This allows
     testers to specify the IP address in cases where there is more than one
     configured IP address for the test system.
     """
-    try:
-        ip = os.environ[LOCAL_IP_ENV]
-    except KeyError:
-        # No env variable set; try to auto detect.
-        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-        s.connect(("8.8.8.8", 80))
-        ip = s.getsockname()[0]
-        s.close()
+    env = LOCAL_IPv6_ENV if v6 else LOCAL_IP_ENV
```

Comment:
I assume there is a minimum IPv6 setup required on your dev box for this to work.  It would be nice to raise an appropriate exception if it appears your dev box is not set up for IPv6 - so that there is a clear user action.  WDYT?